src/Plugin/Action/AddVariable.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\AddVariable.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\rules\Engine\RulesActionBase;
+
+/**
+ * @Action(
+ *   id = "rules_add_variable",
+ *   label = @Translation("Add a variable"),
+ *   context = {
+ *     "value" = @ContextDefinition("any",
+ *       label = @Translation("Value")
+ *     ),
+ *   },
+ *   provides = {
+ *     "variable_added" = @ContextDefinition("any",
+ *        label = @Translation("Added variable")
+ *      )
+ *    }
+ * )
+ */
+class AddVariable extends RulesActionBase {
```

Comment:
You should add a category to the action and prefix the class name with the category - as other classes do.